File: Design-Patterns/Bridge/Cpp/human.cpp
#include "character.h"

human::human(int id, std::string name, int health, int mana) : character(id, name, health, mana)
{

}

void human::set_spell(std::shared_ptr<spell> spell)
{
	character::set_spell(spell);
	spell_->cost -= 10;
	if (spell_->cost < 0) spell_->cost = 0;
}

void human::cast_spell()
{
	character::cast_spell();
	character::cast_spell();
}

File: Design-Patterns/Bridge/CSharp/Bridge/Characters/Human.cs
using Bridge.Spells;

namespace Bridge.Characters
{
    public class Human : Character
    {
        public Human(int id, string name, int health, int mana) : base(id, name, health, mana)
        {

        }

        public override void SetSpell(Spell spell)
        {
            base.SetSpell(spell);
            Spell.Cost -= 10;
            if (Spell.Cost < 0) Spell.Cost = 0;
        }

        public override void CastSpell()
        {
            base.CastSpell();
            base.CastSpell();
        }
    }
}
